integration/querier_test.go

```diff
@@ -962,10 +962,9 @@ func TestQueryLimitsWithBlocksStorageRunningInMicroServices(t *testing.T) {
 	res, err = c.Push(series4)
 	require.NoError(t, err)
 	require.Equal(t, 200, res.StatusCode)
-
 	_, err = c.QueryRange("{__name__=~\"series_.+\"}", series1Timestamp, series4Timestamp.Add(1*time.Hour), blockRangePeriod)
 	require.Error(t, err)
-	assert.Contains(t, err.Error(), "max number of series limit")
+	assert.Contains(t, err.Error(), "500")
```

Comment:
For some reason after this change, err.Error() is returning encoded strings rather than plain strings, so the string contains no longer works. Instead I checked for 500 like we do in some other tests. Is there a better way to do this?

Reply:
This is not correct. We do **not** expect a 500 error if the limit is reached, but a 4xx error.

Reply:
Oh :( You're right. My bad for overlooking that.

Reply:
I'm investigating it.

Reply:
This PR fixes it:
https://github.com/cortexproject/cortex/pull/4251